This adds the necessary bits to make qt attempt building webengine on ppc64.
Since the patch is only sufficient for ppc64le, keep it commented out and
let the packaging template remove the comment later, as there is no good
way to check endianness directly in the file.

We also have to disable jumbo build for the time being because it results
in build failures otherwise, most likely because of compiler miscompiling
the code.
--- qtwebengine/mkspecs/features/functions.prf
+++ qtwebengine/mkspecs/features/functions.prf
@@ -95,6 +95,7 @@
     contains(qtArch, "arm64"): return(arm64)
     contains(qtArch, "mips"): return(mipsel)
     contains(qtArch, "mips64"): return(mips64el)
+    contains(qtArch, "power64"): return(ppc64)
     return(unknown)
 }
 
--- qtwebengine/mkspecs/features/platform.prf
+++ qtwebengine/mkspecs/features/platform.prf
@@ -82,6 +82,7 @@
     contains(QT_ARCH, "arm")|contains(QT_ARCH, "arm64"): return(true)
     contains(QT_ARCH, "mips"): return(true)
 #     contains(QT_ARCH, "mips64"): return(true)
+#ppc64le    contains(QT_ARCH, "power64"): return(true)
 
     skipBuild("QtWebEngine can only be built for x86, x86-64, ARM, Aarch64, and MIPSel architectures.")
     return(false)
--- qtwebengine/src/core/config/linux.pri
+++ qtwebengine/src/core/config/linux.pri
@@ -90,6 +90,10 @@
     else: contains(QMAKE_CFLAGS, "-mdsp"): gn_args += mips_dsp_rev=1
 }
 
+contains(QT_ARCH, "power64") {
+    gn_args += use_jumbo_build=false
+}
+
 host_build {
     gn_args += custom_toolchain=\"$$QTWEBENGINE_OUT_ROOT/src/toolchain:host\"
     GN_HOST_CPU = $$gnArch($$QT_ARCH)
